frontend/src/js/controllers/dashCtrl.js

```diff
@@ -49,12 +52,14 @@
 
         vm.hostChallenge = function() {
 
-            var alert = $mdDialog.alert()
-                .title('Host a challenge')
-                .htmlContent('Please send an email to <a href="mailto:admin@cloudcv.org" class="blue-text">admin@cloudcv.org</a> with the details of the challenge and we will get back soon.')
-                .ok('Close');
-
-            $mdDialog.show(alert);
+            if ($rootScope.isAuth === true)
+            {
+                $state.go('web.challenge-host-teams');
+            }
+            else {
+                utilities.storeData('redirectUrl', 'web.challenge-host-teams');
```

Comment:
Okay, got your point. But don't use local storage for URL storing, there is a concept of save state in angular which memorize the last state of an app. Please go through it and implement that here.